README.md

```diff
@@ -1 +1,177 @@
- # cvs-svc-document-retrieval
+# cvs-svc-document-retrieval
+
+Lambda to retrieve test certificates for commercial vehicles
+
+## Dependencies
+
+The project runs on node 12.x with typescript and serverless framework. For further details about project dependencies, please refer to the `package.json` file.
+[nvm](https://github.com/nvm-sh/nvm/blob/master/README.md) is used to managed node versions and configuration explicitely done per project using an `.npmrc` file.
+
+## Running the project
+
+Once the dependencies are installed, you will be required to rename the `/config/env.example` file to `.env.local` as we use dotenv files for configuration for local local development for example. Further information about [variables](https://www.serverless.com/framework/docs/providers/aws/guide/variables/) and [environment variables](https://www.serverless.com/framework/docs/environment-variables/) with serverless.
+
+The application runs on port `:3001` by default when no stage is provided.
+
+### Environments
+
+We use `NODE_ENV` environment variable to set multi-stage builds (region, stages) with the help of dotenv through npm scripts to load the relevant `.env.<NODE_ENV>` file from `./config` folder into the `serverless.yml` file as we don't rely on serverless for deployment.
+If no `NODE_ENV` value is provided when running the scripts, it will default its `NODE_ENV` value to 'development' with the `.env.development` config file.
+
+The defaulted values for 'stage' and 'region' are `'local'`. Please refer to the values provided in the `serverless.yml` file.
+
+The following values can be provided when running the scripts with `NODE_ENV`:
+
+```ts
+// ./config/.env.<NODE_ENV> files
+'local'; // used for local development
+'development'; // used development staging should we wish to require external services
+'test'; // used during test scripts where local services, mocks can be used in conjonction
+```
+
+```ts
+/** Running serverless offline as an example for a specific stage - 'local'.
+* Stage 'local' will be injected in the serverless.yml
+**/
+NODE_ENV=local serverless offline
+
+```
+
+Further details about environment setup can be found in the provided documentation and `env.example` file.
+
+All secrets the secrets are will stored in `AWS Secrets Manager`.
+
+### Scripts
+
+The following scripts are available, for further information please refer to the project `package.json` file:
+
+- <b>start</b>: `npm start` - _launch serverless offline service_
+- <b>dev</b>: `npm run dev` - _run in parallel the service and unit tests in_ `--watch` _mode with live reload_.
+- <b>test</b>: `npm t` - _execute the unit test suite_
+- <b>build</b>: `npm run build` - _bundle the project for production_
```

Comment:
I think we also have a prod build that will only bundle the relevant dependencies (and not use nodeExternals() from webpack), obfuscate, concatenate, minify that we could test it on a branch and then upload the README with it